File: LC-1342/LC-1342.cpp
#include <cstdlib>

using namespace std;

class Solution {
public:
    /**
     * Concepts: Naive Approach
     */
    int numberOfSteps (int num) {
        int steps = 0;
        while (num > 0) {
            num = (num % 2) ? num - 1 : num / 2;
            ++steps;
        }
        return steps;
    }
};

File: LC-1342/LC-1342.js
/**
 * @param {number} num
 * @return {number}
 */
var numberOfSteps = function (num) {
  /**
   * Concepts: Naive Approach
   */

  let steps = 0
  while (num > 0) {
    num = num % 2 ? num - 1 : num / 2
    ++steps
  }
  return steps
}
